analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py

```diff
@@ -266,30 +313,32 @@ def get_checker_list(self, config) -> Tuple[List[str], List[str]]:
         clang-tidy and in some cases that would cause analysis error due to
         some ClangSA bug.
         """
-        checkers = []
         # Usage of a set will remove compiler warnings and clang-diagnostics
         # which are the same.
+        # In clang-tidy the warnings as compiler-diagnostics are enabled and
+        # disabled by -W<name> and -Wno-<name> respectively. The
+        # clang-diagnostic-<name> is needed, because by default all checkers
+        # are disabled by "checks=-*" and warnings have to be enabled as
+        # checkers with this name: clang-diagnostic-<name>.
         compiler_warnings = set()
+        enabled_checkers = set()
 
         has_checker_config = \
             config.checker_config and config.checker_config != '{}'
 
-        # Do not disable any clang-tidy checks explicitly, but don't run
-        # ClangSA checkers. ClangSA checkers are driven by an other
-        # analyzer in CodeChecker.
-        checkers.append('-clang-analyzer-*')
-
-        # For clang compiler warnings a correspoding
-        # clang-diagnostic error is generated by Clang tidy.
-        # They can be disabled by this glob -clang-diagnostic-*
-        checkers.append('clang-diagnostic-*')
-
         # Config handler stores which checkers are enabled or disabled.
         for checker_name, value in config.checks().items():
             state, _ = value
 
             warning_name, warning_type = \
                 get_compiler_warning_name_and_type(checker_name)
+
+            # This warning must be given a parameter separated by either '=' or
+            # space. This warning is not supported as a checker name so its
+            # special usage is avoided.
+            if warning_name and warning_name.startswith('frame-larger-than'):
```

Comment:
Do we need to this in this patch? Seems to be a new addition. If we must, we could at least emit a curtesy warning that this won't be enabled.

Reply:
This is a function which returns the available checkers. By this `if` statement this "checker" will not be displayed for the users, do they don't even know about the existence of it. They cant provide it in any analyzer command.

Reply:
I might have misunderstood, but in your original comment you suggested to provide a "courtesy warning" for the users which says that this checker/warning will not be enabled. That's why I answered that there is no need for such a warning, because CodeChecker users will not even know about the existence of this "checker" if we exclude it in this `get_checker_list()` function.

And we have to exclude it, because otherwise an invalid `clang-tidy` invocation will be assembled: `clang-tidy ... main.c -- -Wframe-larger-than=`. This is invalid, because this warning requires an integer parameter after `=`. So this is why I decided to hide this checker/warning at all. If I don't exclude this warning then one of our tests fail which enable all "clang-diagnostic-<blabla>" checkers:
`CodeChecker analyze -e clang-diagnostic ...`

Reply:
Aha, I understand now that this function asks clang-tidy for the list of recognized checks, and removed this particular one, so it will act as if the requested didn't exist, resulting in a warning.

But not the _right_ warning. It will say that the check doesn't exist, which is not true -- it does exist, we just don't don't allow running it under CodeChecker. Lets save the user some headache, his Clang binary is not broken -- please emit a warning about this.

warning: CodeChecker is incompatible with compiler warning 'frame-larger-than'. Please don't enable it.